topo/node/cisco/cisco.go

```diff
@@ -148,18 +157,112 @@ func (n *Node) Create(ctx context.Context) error {
 		return fmt.Errorf("failed to create pod for %q: %w", pb.Name, err)
 	}
 	log.Debugf("Pod created:\n%+v\n", sPod)
-	log.Infof("Created XRD node resource %s pod", n.Name())
+	log.Infof("Created Cisco %s node resource %s pod", n.Proto.Model, n.Name())
 	if err := n.CreateService(ctx); err != nil {
 		return err
 	}
-	log.Infof("Created XRD node resource %s services", n.Name())
+	log.Infof("Created Cisco %s node resource %s services", n.Proto.Model, n.Name())
 	return nil
 }
 
+func constraints(pb *tpb.Node) map[string]string {
+	constraints := map[string]string{
+		"cpu":    "1",
+		"memory": "2Gi",
+	}
+	switch pb.Model {
+	case "8201", "8201-32FH", "8202", "8101-32H", "8102-64H":
+		constraints = map[string]string{
+			"cpu":    "4",
+			"memory": "12Gi",
+		}
+	case "xrd":
+		constraints = map[string]string{
+			"cpu":    "1",
+			"memory": "2Gi",
+		}
+	}
+	return constraints
+}
+
+func getCiscoInterfaceId(model string, ethID int) (string, error) {
+	ciscoInterfacePrefix := ""
+	switch model {
+	case "8201":
+		if ethID-1 <= 23 {
+			ciscoInterfacePrefix = "FourHundredGigE0/0/0/"
+
+		} else if ethID-1 <= 35 {
+			ciscoInterfacePrefix = "HundredGigE0/0/0/"
+		}
```

Comment:
so what happens for all interfaces outside of that range?
is that an error or ?

Reply:
I changed the code to return an error and fail early similar to the above case.